freestyle-fs2/shared/src/main/scala/fs2.scala

```diff
@@ -0,0 +1,63 @@
+package freestyle
+
+import _root_.fs2._
+import _root_.fs2.util.{Attempt, Catchable, Free}
+
+import cats._
+import cats.free.{Free => CFree}
+import cats.arrow.FunctionK
+
+import scala.concurrent._
+
+object fs2 {
+  type Eff[A] = Free[CFree[Attempt, ?], A]
+
+  implicit val catsFreeC: Catchable[CFree[Attempt, ?]] = new Catchable[CFree[Attempt, ?]] {
+    def pure[A](a: A): CFree[Attempt, A]                              = CFree.pure(a)
+    def attempt[A](fa: CFree[Attempt, A]): CFree[Attempt, Attempt[A]] = fa.map(Attempt(_))
+    def fail[A](err: Throwable): CFree[Attempt, A]                    = CFree.liftF(Left(err))
+    def flatMap[A, B](a: CFree[Attempt, A])(f: A => CFree[Attempt, B]): CFree[Attempt, B] =
+      a.flatMap(f)
+  }
+
+  implicit val effC: Catchable[Eff] = new Catchable[Eff] {
+    def pure[A](a: A): Eff[A]                            = Free.pure(a)
+    def attempt[A](fa: Eff[A]): Eff[Attempt[A]]          = fa.attempt
+    def fail[A](err: Throwable): Eff[A]                  = Free.fail(err)
+    def flatMap[A, B](a: Eff[A])(f: A => Eff[B]): Eff[B] = a.flatMap(f)
+  }
+
+  @free sealed trait StreamM[F[_]] {
+    def run[A](s: Stream[Eff, A]): FreeS.Par[F, Unit]
+    def runLog[A](s: Stream[Eff, A]): FreeS.Par[F, Vector[A]]
+    def runFold[A, B](z: B, f: (B, A) => B)(s: Stream[Eff, A]): FreeS.Par[F, B]
+    def runLast[A](s: Stream[Eff, A]): FreeS.Par[F, Option[A]]
+  }
+
+  object implicits {
```

Comment:
In this case I can think of an alternative not actually implementing `FreeSLift` since it's too restrictive and given is a fold we can only have syntax when `A` has a Monoid instance.  I think something like this as part of the `implicits` object where the interpreter is now would work to allow `Stream[Eff, A]` to get lifted to `FreeS.Par[F, A]: Thoughts?
```scala
implicit class FS2FreestyleStreamSyntax[A](private val s: Stream[Eff, A]) extends AnyVal {
      def liftFS[F[_]](implicit MA: Monoid[A], SF: StreamM[F]): FreeS[F, A] =
        liftFSPar[F].freeS
      def liftFSPar[F[_]](implicit MA: Monoid[A], SF: StreamM[F]): FreeS.Par[F, A] =
        SF.runFold[A, A](MA.empty, MA.combine)(s)
    }
```